packages/react-components/react-drawer/library/src/shared/drawerMotions.ts

```diff
@@ -17,19 +20,49 @@ const durations: Record<NonNullable<DrawerBaseProps['size']>, number> = {
 /**
  * @internal
  */
-export const InlineDrawerMotion = createPresenceComponent<DrawerMotionParams>(({ position, size }) => {
+export function getPositionTransform(
+  position: DrawerBaseProps['position'],
+  sizeVar: string,
+  dir: FluentProviderContextValue['dir'],
+) {
+  const leftToRightTransform = `translate3d(var(${sizeVar}), 0, 0)`;
+  const rightToLeftTransform = `translate3d(calc(var(${sizeVar}) * -1), 0, 0)`;
+  const bottomToTopTransform = `translate3d(0, var(${sizeVar}), 0)`;
+
+  if (position === 'start') {
+    if (dir === 'rtl') {
+      return rightToLeftTransform;
+    }
+
+    return leftToRightTransform;
```

Comment:
nit: nesting is "bad" and can be mitigated with simple ternary ( yes ternaries are also bad if they go beyond shallow level  - which is not our case)

```suggestion
    return (dir === 'rtl')  ? rightToLeftTransform : leftToRightTransform;
```